tests/core/test_variable.py

```diff
@@ -0,0 +1,558 @@
+        v = relentless.DesignVariable(value=1.0, low=0.5, high=1.5)
+        w = relentless.ArithmeticMean(u, v)
+        x = relentless.ArithmeticMean(v, w)
+        y = relentless.ArithmeticMean(w, x)
+
+        self.assertEqual(u.value, 2.0)
+        self.assertEqual(v.value, 1.0)
+        self.assertAlmostEqual(w.value, 1.5)
+        self.assertAlmostEqual(x.value, 1.25)
+        self.assertAlmostEqual(y.value, 1.375)
+
+        #change value of u
+        u.value = 2.6
+        self.assertEqual(u.value, 2.5)
+        self.assertEqual(v.value, 1.0)
+        self.assertAlmostEqual(w.value, 1.75)
+        self.assertAlmostEqual(x.value, 1.375)
+        self.assertAlmostEqual(y.value, 1.5625)
+
+        #change value of v
+        v.value = 0.4
+        self.assertEqual(u.value, 2.5)
+        self.assertEqual(v.value, 0.5)
+        self.assertAlmostEqual(w.value, 1.5)
+        self.assertAlmostEqual(x.value, 1.0)
+        self.assertAlmostEqual(y.value, 1.25)
+
+    def test_derivative(self):
+        """Test _derivative method."""
+        u = relentless.DesignVariable(value=1.0)
+        v = relentless.DesignVariable(value=2.0)
+        w = relentless.ArithmeticMean(u, v)
+
+        #test w.r.t. a
+        dw = w._derivative('a')
+        self.assertEqual(dw, 0.5)
+
+        #test w.r.t. b
+        dw = w._derivative('b')
+        self.assertEqual(dw, 0.5)
+
+        #test w.r.t. ~a,~b
+        with self.assertRaises(ValueError):
+            w._derivative('c')
+
+class test_GeometricMean(unittest.TestCase):
+    """Unit tests for relentless.GeometricMean"""
+
+    def test_init(self):
+        """Test creation with data."""
+        #test variable dependent on 2 DesignVariables
+        u = relentless.DesignVariable(value=1.0)
+        v = relentless.DesignVariable(value=2.0)
+        w = relentless.GeometricMean(u, v)
+        self.assertAlmostEqual(w.value, 1.4142136)
+        self.assertCountEqual(w.params, ('a','b'))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'a':u,'b':v})
+
+        #test variable dependent on 1 DesignVar, 1 DependentVar
+        x = relentless.GeometricMean(v,w)
+        self.assertAlmostEqual(x.value, 1.6817928)
+        self.assertCountEqual(x.params, ('a','b'))
+        self.assertDictEqual({p:v for p,v in x.depends}, {'a':v,'b':w})
+
+        #test variable dependent on 2 DependentVariables
+        y = relentless.GeometricMean(w,x)
+        self.assertAlmostEqual(y.value, 1.5422108)
+        self.assertCountEqual(y.params, ('a','b'))
+        self.assertDictEqual({p:v for p,v in y.depends}, {'a':w,'b':x})
+
+        #test same-valued dependencies
+        z = relentless.DesignVariable(value=1.0)
+        w = relentless.GeometricMean(u, z)
+        self.assertAlmostEqual(w.value, 1.0)
+        self.assertCountEqual(w.params, ('a','b'))
+        self.assertDictEqual({p:v for p,v in w.depends}, {'a':u,'b':z})
+
+        #test same-object dependencies
+        w = relentless.GeometricMean(u, u)
```

Comment:
Can you do this test with v rather than u, so that it actually multiplies to a nontrivial value?